oauth2/src/main/java/com/okta/spring/boot/oauth/env/OIDCMetadata.java

```diff
@@ -0,0 +1,85 @@
+package com.okta.spring.boot.oauth.env;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.http.ResponseEntity;
+
+public class OIDCMetadata {
+
+    private boolean isAuth0;
+    private final String clientAuthenticationMethod = "none";
+    private final String scope = "profile,email,openid";
+    private final String jwkSetURI;
+    private final String authorizationURI;
+    private final String tokenURI;
+    private final String userInfoURI;
+    private final String introspectionURI;
+
+    public boolean isAuth0() {
```

Comment:
I landed here, but I know nothing about this project. However, I couldn't help but notice this `isAuth0` flag, which reminds me a bit of `isIE` from many years ago 🙂 
I'm not sure what this flag is for, but wouldn't it be better to do a feature check instead of a product check? Product features may change over time
Sorry for the intrusion 🙂

Reply:
No worries @andychiare and thank you. This is to differentiate between auth0 domains and okta domains. How do you propose the feature test?

Reply:
It depends on why you need to differentiate between Auth0 and Okta domains.
For example, if you need this differentiation because Okta supports the introspection endpoint while Auth0 doesn't, check directly if the introspection endpoint is missing.
If and when Auth0 implements its introspection endpoint, you will not need to change your code 🙂 
Does this make sense?

Reply:
That was my original implementation, but then there was another place where I wanted to check if it's an auth0 domain to exclude it from some url manipulation done for okta domains. But I guess you are right that looking for the url alone here might be better in the long run. I'll update this